src/fill/alloy.rs

```diff
@@ -306,46 +325,27 @@ impl Tx for alloy::rpc::types::TransactionRequest {
 
         // NB: this is set to max if not provided, as users will typically
         // intend that to mean "as much as possible"
+        *tx_type = self.transaction_type.unwrap_or(TxType::Eip1559 as u8);
         *gas_limit = self.gas.unwrap_or(u64::MAX);
-        *gas_price = U256::from(self.gas_price.unwrap_or_default());
-        *transact_to = self.to.unwrap_or_default();
+        *gas_price = self.gas_price.unwrap_or_default();
+        *kind = self.to.unwrap_or_default();
         *value = self.value.unwrap_or_default();
         *data = self.input.input().cloned().unwrap_or_default();
-        *nonce = self.nonce;
+        *nonce = self.nonce.unwrap_or_default(); // TODO: IS THIS CORRECT?
```

Comment:
hmm i'm not exactly sure this is correct, i need to check

Reply:
soooo it's not the same behavior as it used to be, unfortunately. Because nonce-check-disabling is now in cfg rather than tx. but there's nothing else we can do here -_-

Reply:
the rpc server already disables nonce checks on call so this should be fine